src/NHibernate/Loader/Loader.cs

```diff
@@ -778,23 +778,25 @@ protected virtual object[] GetResultRow(Object[] row, DbDataReader rs, ISessionI
 		/// </summary>
 		private void RegisterNonExists(EntityKey[] keys, ISessionImplementor session)
 		{
-			int[] owners = Owners;
-			if (owners != null)
+			var owners = Owners;
```

Comment:
>For missing objects associated by one-to-one with...

Comment for method should be adjusted as it's not one-to-one specific anymore.

Same applies for protected `Owners` property:
>An array of indexes of the entity that owns a one-to-one association

For one-to-one - null key is possible only for not-found ignore mapping, but for many-to-one we may have many valid null keys. So I it would be good to improve this place to call AddNullProperty only if association is mapped as not-found ignore. Just a thought..

Reply:
I've updated comments.

>it would be good to improve this place to call AddNullProperty only if association is mapped as not-found ignore. Just a thought..

It actually does not matter. I think it saves some calls in `ResolveIdentifier`. However, I think we need to add some property-ref tests.

Reply:
>It actually does not matter.

I meant only wasting memory. This check would save some.

Reply:
Actually it seems check to avoid unnecessary filling dictionary with null keys is already available via `ownerAssociationType.IsNullable`. I think we should add it here